### PR TITLE
Add max length validation

### PR DESCRIPTION
**Pull request by awaisdar001**
I want to fix and sync course metadata model `Course` and publisher model `Course`. The issue is, when we `publish` a course from publisher, it copies the data to course metadata course model and it is very important that both models should be in sync in terms of `max_length` validations. 

I have removed the old `validate_text_count` validator b/c the approch it was using doesn't make any sense to count text only when you are saving the html in the model. Instead, I used Djangos' own `MaxLengthValidator`. 

- Course Metadata Model ([Course](https://github.com/edx/course-discovery/blob/b4e864145902f4b21ce54a96aca074a8451c4e42/course_discovery/apps/course_metadata/models.py#L337)): 
- Publisher Model ([Course](https://github.com/edx/course-discovery/blob/2c7ebc6aafa92907e30e6544ecdc5e98bbb6e372/course_discovery/apps/publisher/models.py#L44)):

[EDUCATOR-3315](https://openedx.atlassian.net/browse/EDUCATOR-3315)

### CODE REVIEW

File: course_discovery/apps/publisher/forms.py [hunk 1]
```diff
@@ -75,22 +76,24 @@ class CourseForm(BaseForm):
         label=_('Organization Name'),
         required=True
     )
-    title = forms.CharField(label=_('Course Title'), required=True)
+    title = forms.CharField(label=_('Course Title'), required=True, validators=[MaxLengthValidator(255)])
```
Comment (awaisdar001) [inline course_discovery/apps/publisher/forms.py]: Note for reviewers: I add `255` length constraint to match with [course metadata course model title](https://github.com/edx/course-discovery/blob/b4e864145902f4b21ce54a96aca074a8451c4e42/course_discovery/apps/course_metadata/models.py#L345) constraint.

File: course_discovery/apps/publisher/forms.py [hunk 2]
```diff
@@ -75,22 +76,24 @@ class CourseForm(BaseForm):
...
         label=_('Course Number'), required=True,
-        validators=[validate_text_count(max_length=50)]
+        validators=[MaxLengthValidator(50)]
     )
     short_description = forms.CharField(
         label=_('Short Description'),
-        widget=forms.Textarea, required=False, validators=[validate_text_count(max_length=255)]
+        widget=forms.Textarea,
+        required=False,
+        validators=[MaxLengthValidator(350)]
```
Comment (awaisdar001) [inline course_discovery/apps/publisher/forms.py:90]: Same for this one. Matching with  [course metadata course model short_description](https://github.com/edx/course-discovery/blob/b4e864145902f4b21ce54a96aca074a8451c4e42/course_discovery/apps/course_metadata/models.py#L346) constraint.

File: course_discovery/static/js/publisher/tinymce-charactercount.plugin.js [hunk 3]
```diff
@@ -30,16 +30,7 @@ tinymce.PluginManager.add('charactercount', function (editor) {
   });
 
   _self.getCount = function () {
-    var tx = editor.getContent({ format: 'raw' });
-    var decoded = decodeHtml(tx);
-    var decodedStripped = decoded.replace(/(<([^>]+)>)/ig, "").trim();
```
Comment (awaisdar001) [inline course_discovery/static/js/publisher/tinymce-charactercount.plugin.js:35]: Similarly, Do not need to remove `html` elements to show counts, be explicit to the user about how much the data length is.

File: course_discovery/static/js/publisher/tinymce-init.js [hunk 4]
```diff
@@ -77,6 +77,9 @@ $(document).ready(function () {
                     }
                 }
             });
+            editor.on('GetContent', function (e) {
```
Comment (awaisdar001) [inline course_discovery/static/js/publisher/tinymce-init.js:80]: Basically, Just to match the Django `max_length` constraint I removed `\n` and `\r` from the content.

**Review by noraiz-anwar — APPROVED** [review]
Nice cleanup! 
👍

**Review by asadazam93 — APPROVED** [review]
Looks good

**Pull request merged**